ecs-cli/integ/cmd_up/up_test.go

```diff
@@ -0,0 +1,289 @@
+// +build integ
+
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Package cmd_up upTests the "ecs-cli up" command with various configurations.
+package cmd_up
+
+import (
+	"fmt"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/integ"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+const (
+	// timeoutForWaitingOnActiveInstancesInS is how long we are willing wait
+	// for the container instances to become ACTIVE in the cluster.
+	timeoutForWaitingOnActiveInstancesInS = 300
+
+	// sleepDurationInBetweenRetriesInS is how long we sleep in between retrying requests that fail.
+	sleepDurationInBetweenRetriesInS = 30
+)
+
+type upTest struct {
+	clusterName                 string
+	cmdArgs                     []string
+	wantedStdoutSnippets        []string
+	wantedClusterSize           int
+	wantedInstanceAttributes    []*ecs.Attribute
+	wantedInstanceResourceNames []string
+}
+
+var upTests = []upTest{
+	{
+		// Test cluster creation with default configurations
+		integ.SuggestedResourceName("TestCmdUpWithDefaultConfig"),
+		[]string{"up", "--capability-iam"},
+		[]string{
+			"Using recommended Amazon Linux 2 AMI",
+			"VPC created",
+			"Security Group created",
+			"Subnet created",
+			"Cluster creation succeeded",
+		},
+		1,
+		[]*ecs.Attribute{
+			{
+				Name:  aws.String("ecs.instance-type"),
+				Value: aws.String("t2.micro"),
+			},
+		},
+		[]string{},
+	},
+	{
+		// Test cluster creation with a GPU instance
+		integ.SuggestedResourceName("TestCmdUpWithGPUInstance"),
+		[]string{"up", "--instance-type", "p2.xlarge", "--capability-iam"},
+		[]string{
+			"Using GPU ecs-optimized AMI because instance type was p2.xlarge",
+			"VPC created",
+			"Security Group created",
+			"Subnet created",
+			"Cluster creation succeeded",
+		},
+		1,
+		[]*ecs.Attribute{
+			{
+				Name:  aws.String("ecs.instance-type"),
+				Value: aws.String("p2.xlarge"),
+			},
+		},
+		[]string{"GPU"},
+	},
+}
+
+// TestCmd_UP runs the 'ecs-cli up [command options] [arguments...]' command
+// given a list of configurations and expected results.
+func TestCmd_UP(t *testing.T) {
+	cfnClient, ecsClient := setup(t)
+	for _, test := range upTests {
```

Comment:
Can we run each case as a subtest and log out a description? See this [example](https://github.com/aws/amazon-ecs-cli/blob/master/ecs-cli/modules/cli/compose/adapter/convert_test.go#L662).